### PR TITLE
Add hardware timestamp reporting

### PR DESCRIPTION
**Pull request by C-Achard**
## Summary

Adds frame capture metadata and recording timestamp support, with best-effort Basler hardware timestamp integration.

This PR refactors camera reads to allow returning extra objects alongside the frame, such as the existing host-side software timestamp, as well as optional backend-provided timestamp metadata. 
The software timestamp remains the primary source, while hardware timestamps are added only when the backend can report them.

## Main changes

- Introduces a CapturedFrame wrapper for camera backends
- Adds generic timestamp metadata support
  - Supports hardware camera-clock timestamps, raw device ticks, optional wall-clock timestamps, and backend-specific extras
- Adds best-effort Basler hardware timestamp support
  - Reads device ticks count reported by Basler grab results when available
  - Falls back to an assumed one gigahertz tick frequency when the camera does not report one, and records that assumption in metadata. See [Basler docs](https://docs.baslerweb.com/timestamp#how-it-works), generally devices which cannot report tick frequency operate at 1GHz, hence the fallback.
  - Treats zero-valued Basler timestamps as unavailable, which handles cases such as pylon emulation or cameras that expose the API but do not provide usable timestamps, as per Basler docs.
- Updates non-Basler backends to return the new captured-frame object.
  - Aravis, GenTL, and OpenCV currently preserve software timestamps and do not provide hardware timestamp metadata for now.
- Threads timestamp metadata through the recording path.
  - RecordingManager forwards timestamp metadata into VideoRecorder.
- Updates recorder timestamp sidecar output.
  - Timestamp sidecars now use schema version 2.
  - Per-recording timestamp source metadata is stored once.
  - Hardware timestamp fields are omitted when unavailable.
- Improves Basler exposure compatibility
  - Sets both ExposureTime and ExposureTimeAbs when writable as they both control the same property on different devices

## What this does not do yet

- Hardware timestamps are best effort and backend-dependent.
- This does not change recording throughput or move recorder enqueueing off the GUI thread.

## Testing

Added and updated tests accordingly; tested and confirmed to work as intended with 1x a2A1920-160umPRO.

### CODE REVIEW

File: dlclivegui/cameras/base.py [hunk 1]
```diff
@@ -107,6 +124,11 @@ def actual_pixel_format(self) -> str | None:
     def recommended_preserve_mono(self) -> bool | None:
         return None
 
+    @property
+    def last_frame_timestamp_metadata(self) -> FrameTimestampMetadata | None:
+        """Return backend-provided timestamp metadata for the last read frame."""
+        return None
+
```
Comment (deruyter92) [inline dlclivegui/cameras/base.py]: currently nothing seems to override this or use it, are you planning to extend this? otherwise perhaps drop it.

File: dlclivegui/cameras/backends/basler_backend.py [hunk 2]
```diff
@@ -678,7 +703,36 @@ def open(self) -> None:
...
-    def read(self) -> tuple[np.ndarray, float]:
+    def _make_timestamp_metadata(self, grab_result) -> FrameTimestampMetadata | None:
+        try:
+            ticks = int(grab_result.GetTimeStamp())
+        except Exception:
+            return None
+
+        if ticks == 0:
+            # Basler returns 0 if the timestamp is not available (e.g. for some GigE cameras)
+            return None
```
Comment (deruyter92) [inline dlclivegui/cameras/backends/basler_backend.py:713]: maybe add a unit test for zero ticks case or missing tick frequency. Currently only the happy path seems tested.

File: dlclivegui/cameras/base.py (hunk 1 above)
Comment (C-Achard) [reply dlclivegui/cameras/base.py]: Dropped it, thanks!
